app/controllers/api/v1/services_controller.rb

```diff
@@ -0,0 +1,69 @@
+module Api
+  module V1
+    class ServicesController < ApplicationController
+      before_action :set_service, only: %i[show update destroy]
+
+      attr_accessor :service
+
+      def index
+        services = Service.all
+        render json: services
+      end
+
+      def show
+        render json: service if service
```

Comment:
Codesmell - you should make sure that service is present before action is invoked. Otherwise you should let client know that there is no such service with supplied ID.

Reply:
Condition is useless here. There is before_action :set_service which raise RecordNotFound exception if there is no record with supplied ID. The exception is handled in application_controller